src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
That's right. If you use `cmark_node_new`, you can only add that node to a document created from `cmark_parser_new` (since these APIs use the default system allocator)... Mixing and matching nodes is not a good idea. A library user should always use the same allocator, hence this check is important!